pxr/base/tf/anyWeakPtr.h

```diff
@@ -174,9 +168,7 @@ class TfAnyWeakPtr : boost::totally_ordered<TfAnyWeakPtr>
         TF_API virtual TfWeakBase const *GetWeakBase() const;
         TF_API virtual operator bool() const;
         TF_API virtual bool _IsConst() const;
-#ifdef PXR_PYTHON_SUPPORT_ENABLED
-        TF_API virtual boost::python::api::object GetPythonObject() const;
-#endif // PXR_PYTHON_SUPPORT_ENABLED
+        TF_API virtual TfPyBoostObjWrapper GetPythonObject() const;
```

Comment:
The other difference between the two uses is is that TfPyObjWrapper is holding a shared_ptr to an object.  However, object is basically a reference-counted holder as well, so it should have the same semantic.

Reply:
> The other difference between the two uses is is that TfPyObjWrapper is holding a shared_ptr to an object. However, object is basically a reference-counted holder as well, so it should have the same semantic.

 The big important reason why this class exists and uses a shared_ptr is that in order to use boost::python::object you have to be holding the python GIL, since it invokes the python ref-counting operations under the hood, and boost::python::object does not take the GIL itself.  This is explained in the class doc for TfPyObjWrapper (https://graphics.pixar.com/usd/release/api/class_tf_py_obj_wrapper.html#details)

> Provides a wrapper around boost::python::object that works correctly for the following basic operations regardless of the GIL state: default construction, copy construction, assignment, (in)equality comparison, hash_value(), and destruction.
>
> None of those work correctly in the presence of an unlocked GIL for boost::python::object. This class only actually acquires the GIL for default construction, destruction and for some (in)equality comparisons. The other operations do not require taking the GIL.

Reply:
It is worth noting that in the latest commit, in which we use only the wrapper class TfPyObjWrapper,  there will be a small overhead when creating the TfPyObjWrapper and deleting it.  For example:

    `TfPyObjWrapper. TfAnyWeakPtr::_PointerHolder<Ptr>::GetPythonObject() `

will return a `TfPyObjWrapper` and a subsequent call to the `Get() `method of a `TfPyObjWrapper` object will return the `boost::python::object`.

Thus, on the one hand, there is small cost when going through `TfPyObjWrapper`, and, on the other hand, it is protected because of the pointer management of `TfPyObjWrapper`. 

Is that OK?